Disable the pre-sales chat on the Cloud pricing page for the Holidays.

elliottprogrammer opened the pull request:
#### Proposed Changes

This PR disables the Zendesk pre-sales Chat interface on the Jetpack Cloud pricing page automatically between the dates of Dec 23, 2022 midnight UTC **_thru_** Jan 3, 2023 midnight UTC.

Related to: p1671473237696439-slack-C04BZQFCWE9

#### Testing Instructions

Note: The pre-sales chat does not run on weekends (UTC), so if you're testing this PR on a weekend day and the chat is not showing, this is why.  You'll need to change your local computer date/time to a non-weekend day.

- Do any one of these
    * Click on Jetpack Cloud live link below and wait for the redirect to complete, then goto `/pricing`
    * or boot up this PR 
        * Run `git fetch && git checkout update/disable-presales-chat-for-holidays`
        * Run `yarn start-jetpack-cloud`
        * Goto [http://jetpack.cloud.localhost:3000/pricing](http://jetpack.cloud.localhost:3000/pricing)
- Assuming you are **not** testing this between the shutdown dates (listed above), verify you see the chat box in the lower right corner of the page.
- Now change your local computer's clock to some time(s) between the shutdown dates (listed above). 
    - One way to test would be to go to worldtimebuddy.com and create a row with your own location and then create another row with UTC location. Then find what time it is for you at Dec 23, 2022 midnight UTC and also at Jan 2, 2023 midnight UTC. Write down or keep note of these local date/times.
    - Then change your local computer time to a time just _after_ the **start** shutdown time for you locally. Reload the page and verify the chat is not showing.
    - set your local computer date/time to a time just _before_ the **end** shutdown time, and verify the chat is still not showing.
    - Then set your local computer date/time to a time just _after_ the **end** shutdown time for you locally. Verify the chat is now showing again.


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

elliottprogrammer committed:
Shut down chat from 12-23-2022 thru 01-03-2023 UTC.

TheSteveK reviewed (dismissed):
I was having some issues with the chat still appearing, so I logged the variables for isWithinShutdownDates and got something unexpected:

![Screenshot 2022-12-21 at 7 00 04 AM](https://user-images.githubusercontent.com/12895386/208900772-fb898973-3b74-4099-9198-dacc53cc9d03.png)

mattgawarecki commented:
> I was having some issues with the chat still appearing, so I logged the variables for isWithinShutdownDates and got something unexpected:
> 
> ![Screenshot 2022-12-21 at 7 00 04 AM](https://user-images.githubusercontent.com/12895386/208900772-fb898973-3b74-4099-9198-dacc53cc9d03.png)

Ah, nice catch! Though I'll call it out explicitly, because it took me a minute to find: the start time is set to **January** 22 of next year, rather than **December** of this year.

elliottprogrammer committed:
Fix start date, as per feedback.

elliottprogrammer committed:
Fix startDate and endDate to match/sync with Jetpack.com,
so that the chat shutdown occurs at the same time in both Calypso and Jetpack.com.

elliottprogrammer committed:
Oops, change endDate from Jan 3, to Jan 2.

elliottprogrammer committed:
Fix the comment to reflect the correct EST time.

elliottprogrammer commented:
Thanks @TheSteveK and @mattgawarecki for catching this. 👍

It's strange.. I actually encountered this bug already when I was coding it.. (it's when I was reminded/re-learned that JavaScript _months_ go from 0 to 11). I could swear I fixed both the _startTime_ and _endTime_, but apparently I did something whereas the _startTime_ didn't get fixed properly. 🤷‍♂️   Anyway, thanks again, got it all fixed now.

Also I fixed the _endTime_ to match the end date on Jetpack.com (Jan 2).

elliottprogrammer commented:
Wait a minute.. I just realized a problem with this PR.. I need to move the holiday shutdown logic out from this component because I realize this component is used elsewhere also, I believe.. I'm labeling this PR "in progress" for the moment.

elliottprogrammer commented:
> Wait a minute.. I just realized a problem with this PR.. I need to move the holiday shutdown logic out from this component because I realize this component is used elsewhere also, I believe.. I'm labeling this PR "in progress" for the moment.

No, I was wrong about this..  ^^
This is good.  It needs to be deployed.

andrii-lysenko reviewed (approved):
LGTM! If the timeframes what we desire, the code looks good.
